code/game/objects/structures/crates_lockers/largecrate.dm

```diff
@@ -128,6 +128,23 @@
 	desc = "Two small black storage cases."
 	icon_state = "case_small"
 
+/obj/structure/largecrate/random/barrel/attackby(obj/item/tool/weldingtool/W as obj, mob/user as mob)
+	if(W.welding == 1)
+		if(do_after(user, 50, TRUE, 5, BUSY_ICON_BUILD))
+			new /obj/item/stack/sheet/metal/small_stack(src)
+			W.remove_fuel(1,user)
+			var/turf/T = get_turf(src)
+			for(var/obj/O in contents)
+				O.loc = T
+			user.visible_message("<span class='notice'>[user] welds \the [src] open.</span>", \
```

Comment:
Add a little sound?
`	playsound(src.loc, 'sound/items/Welder2.ogg', 25, 1)`
Check the barricade repair code for an example.